glib/meson.build

```diff
@@ -52,21 +52,13 @@ base_include_directories = [
 
 gi_fatal_warnings = (build_machine.system() != 'windows')
 
-adbc_build_dir = get_option('adbc_build_dir')
-if adbc_build_dir != ''
-  adbc_build_dir = meson.source_root() / adbc_build_dir
-endif
-
-if adbc_build_dir == ''
-  adbc_driver_manager = dependency('adbc-driver-manager')
```

Comment:
This is for building `.deb`.
To build `.deb` from one source archive, we can't assume that `c/` is already installed.

https://github.com/apache/arrow-adbc/blob/66ecd33dceb9b8f9fe513abf99685e21929a7064/ci/linux-packages/debian/rules#L33-L65

Why should we remove this? If we want to remove this, we should use only Meson to build `.deb`. Should we do it?

Reply:
Yea I noticed Meson was being built with the --nofallback wrap option so assumed it would need to find the driver manager lib in the system to work. 

Thanks for the link - will give it a read. Definitely not a topic I am familiar with